packages/next/src/build/webpack/loaders/next-app-loader.ts

```diff
@@ -116,32 +118,30 @@ async function createAppRouteCode({
     nextConfigOutput: ${
       nextConfigOutput ? JSON.stringify(nextConfigOutput) : 'undefined'
     },
-    requestAsyncStorage,
-    staticGenerationAsyncStorage,
-    staticGenerationBailout,
-    headerHooks,
-    serverHooks,
   }`
 
   return `
     import 'next/dist/server/node-polyfill-headers'
 
-    import { Route } from 'next/dist/build/webpack/loaders/next-app-loader/routes/${kind}'
+    import RouteModule from 'next/dist/server/future/route-modules/${kind}/module'
 
-    import { requestAsyncStorage } from 'next/dist/client/components/request-async-storage'
-    import { staticGenerationAsyncStorage } from 'next/dist/client/components/static-generation-async-storage'
+    import * as userland from ${JSON.stringify(resolvedPagePath)}
 
-    import * as headerHooks from 'next/dist/client/components/headers'
-    import * as serverHooks from 'next/dist/client/components/hooks-server-context'
-    import { staticGenerationBailout } from 'next/dist/client/components/static-generation-bailout'
+    const routeModule = new RouteModule(${options})
 
-    import * as userland from ${JSON.stringify(resolvedPagePath)}
+    // Pull out the exports that we need to expose from the module. This should
+    // be eliminated when we've moved the other routes to the new format. These
+    // are used to hook into the route.
+    const {
+      requestAsyncStorage,
+      staticGenerationAsyncStorage,
+      serverHooks,
+      headerHooks,
+      staticGenerationBailout
+    } = routeModule
```

Comment:
We still have to expose these from the module (as named exports) because of all the code accross the server that interacts with them (for example ISR) assumes these are named exports.